internal/controllers/image_controller.go

```diff
@@ -223,17 +223,44 @@ func (r *ImageReconciler) deleteImage(ctx context.Context, log logr.Logger, ioCt
 		return nil
 	}
 
-	if err := librbd.RemoveImage(ioCtx, ImageIDToRBDID(image.ID)); err != nil && !errors.Is(err, librbd.ErrNotFound) {
-		return fmt.Errorf("failed to remove rbd image: %w", err)
+	pools, imgs, err := listRbdImageChildren(ioCtx, ImageIDToRBDID(image.ID))
+	log.V(2).Info("Volume image references", "pools", pools, "rbd-images", imgs)
+
+	if err != nil {
+		return fmt.Errorf("failed to list volume image children: %w", err)
 	}
-	log.V(2).Info("Rbd image deleted")
+	if pools == 0 && imgs == 0 {
```

Comment:
Let's have an early return by inverting this if statement:


```suggestion
	if pools != 0 || imgs != 0 {
	    return nil
	}
```

Then we don't have to indent the rest of the method.